### DISCUSSION TITLE
Trigger notes independently?

### DISCUSSION THREAD

**Discussion by ardenschager** [chaosprint/glicol#128] [General]
Hello chaosprint! I love playing with glicol. The performance potential and ease of use have me quite excited

I am aware this framework is mostly for live-coding, and will certainly be messing with it for that, but I was also wondering if there would be support for midi or osc sometime soon? I would love to basically use this as a replacement for tone.js for certain use cases. I like to do generative stuff with p5.js that is arrhythmic / polyphonic, with independent elements triggering voices, and often that quickly runs into performance issues on web, at least on Tone.js without audioworklets. Seems like you have done a lot of the much-needed optimization already for this library. 

TLDR, it would be great to be able to trigger a polyphonic synth using some kind of midi/osc command. Maybe there a way to do that now? 

Best and thanks!

**Comment by chaosprint** [comment]
Thanks for the post.
I have marked them as todo here (https://github.com/chaosprint/glicol/issues/35) but it's not yet implemented.
The main concern is that Glicol is signal-centred (everything is signal) while MIDI and OSC are more towards logic events.
I am actually thinking about having another more general language that wraps both but there are too many tasks in front of me now. 
Meanwhile if the browser environment is not a necessity for you, perhaps you can also try https://github.com/supercollider/supercollider

**Reply by ardenschager** [reply]
Thanks for the response! That makes sense. Just wanted to throw this request out there, the browser environment actually is a big necessity for me which is why I'm excited for this library. In the mean time, I'll just look into the existing examples some more.
